File: go-client-sdk/pkg/models/operations/getcompaniescompanyiddatadatasetlogsdatasetid.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetCompaniesCompanyIDDataDatasetLogsDatasetIDPathParams struct {
	CompanyID string `pathParam:"style=simple,explode=false,name=companyId"`
	DatasetID string `pathParam:"style=simple,explode=false,name=datasetId"`
}

type GetCompaniesCompanyIDDataDatasetLogsDatasetIDSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetCompaniesCompanyIDDataDatasetLogsDatasetIDRequest struct {
	PathParams GetCompaniesCompanyIDDataDatasetLogsDatasetIDPathParams
	Security   GetCompaniesCompanyIDDataDatasetLogsDatasetIDSecurity
}

type GetCompaniesCompanyIDDataDatasetLogsDatasetIDResponse struct {
	CodatPullMessagesContractsDatasetMessages *shared.CodatPullMessagesContractsDatasetMessages
	ContentType                               string
	StatusCode                                int
}

File: typescript-client-sdk/src/sdk/models/operations/getcompaniescompanyiddatadatasetlogsdatasetid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetCompaniesCompanyIdDataDatasetLogsDatasetIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=datasetId" })
  datasetId: string;
}


export class GetCompaniesCompanyIdDataDatasetLogsDatasetIdSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeApiKeyAuth;
}


export class GetCompaniesCompanyIdDataDatasetLogsDatasetIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCompaniesCompanyIdDataDatasetLogsDatasetIdPathParams;

  @SpeakeasyMetadata()
  security: GetCompaniesCompanyIdDataDatasetLogsDatasetIdSecurity;
}


export class GetCompaniesCompanyIdDataDatasetLogsDatasetIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatPullMessagesContractsDatasetMessages?: shared.CodatPullMessagesContractsDatasetMessages;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
